examples/some-issuer/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use anyhow::Context;
+use axum::Json;
+use axum_sessions::async_session::chrono::{self, TimeZone};
+use concordium_rust_sdk::cis4::{Cis4Contract, Cis4TransactionMetadata};
+use concordium_rust_sdk::common::types::{KeyPair, TransactionTime};
+use concordium_rust_sdk::contract_client::CredentialInfo;
+use concordium_rust_sdk::id::constants::{ArCurve, AttributeKind};
+use concordium_rust_sdk::id::pedersen_commitment;
+use concordium_rust_sdk::smart_contracts::common::{Amount, Duration, Timestamp};
+use concordium_rust_sdk::types::hashes::TransactionHash;
+use concordium_rust_sdk::types::transactions::send::GivenEnergy;
+use concordium_rust_sdk::types::{CryptographicParameters, Energy, Nonce, WalletAccount};
+use concordium_rust_sdk::web3id::did::Network;
+use concordium_rust_sdk::web3id::{SignedCommitments, Web3IdAttribute, Web3IdCredential};
+use reqwest::{StatusCode, Url};
+use serde::Serialize;
+
+#[derive(Clone)]
+pub struct IssuerState {
+    pub crypto_params: Arc<CryptographicParameters>,
+    pub contract_client: Cis4Contract,
+    pub network: Network,
+    pub issuer: Arc<WalletAccount>,
+    pub issuer_key: Arc<KeyPair>,
+    pub nonce_counter: Arc<tokio::sync::Mutex<Nonce>>,
+    pub max_register_energy: Energy,
+    pub metadata_url: Arc<Url>,
+    pub credential_schema_url: Arc<Url>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IssueResponse {
+    tx_hash: TransactionHash,
+    credential: Web3IdCredential<ArCurve, Web3IdAttribute>,
+}
+
+#[tracing::instrument(level = "info", skip(issuer, credential, user_id))]
+pub async fn issue_credential(
+    issuer: IssuerState,
+    credential: CredentialInfo,
+    user_id: String,
+) -> Result<Json<IssueResponse>, StatusCode> {
+    tracing::info!("Request to issue a credential.");
+
+    if let Err(err) = validate_credential(&issuer, &credential) {
+        tracing::warn!("Failed to validate credential: {err}");
+        return Err(StatusCode::BAD_REQUEST);
+    }
+
+    match register_credential(issuer, credential, user_id).await {
+        Ok(res) => {
+            tracing::info!("Successfully issued credential.");
```

Comment:
Some more details about the credential should be here.

And this should be at debug I think.

Reply:
In the version I am working on it prints the user id and username, is that the kind of thing you were referring to?

> And this should be at debug I think.

I assume the corresponding "Request to issue a credential." and "" Using nonce... to send transaction." also should be debug then.

Reply:
Yes.